src/main/java/net/revelc/code/impsort/ImpSort.java

```diff
@@ -166,4 +178,55 @@ private static void convertAndAddImport(LinkedHashSet<Import> allImports, List<N
     allImports.add(imp);
   }
 
+  /*
+   * Extract all of the tokens from the main body of the file.
+   *
+   * This set of tokens represents all of the file's dependencies, and is used
+   * to figure out whether or not an import is unused.
+   */
+  private static Set<String> extractTokens(CompilationUnit unit) {
+    Set<String> tokens = new LinkedHashSet<String>();
+    NodeList<TypeDeclaration<?>> types = unit.getTypes();
+    types.forEach((Node node) -> {
```

Comment:
Out of curiosity, why was the explicit "Node" type necessary here in this lambda?

Reply:
I'll make a wild guess that `<?>` was throwing the type inference off.

Reply:
Actually, I tested without the explicit type, and it works. I'm not sure why I added it, but it was probably because I thought it wouldn't work with the `<?>` otherwise. I can remove this type.

Reply:
It doesn't matter to me. I was just curious. I have only just begun to learn all the quirks of lambdas in Java :smiley_cat: